Limiting plugin types 2 (#150)
ansible-collections/overview: collection_requirements.rst

Modules & Plugins
------------------

* Collections MUST only use the directories specified below in the ``plugins/`` directory and
  only for the purposes listed:

  :Those recognized by ansible-core: ``doc_fragments``, ``modules``, ``module_utils``, ``terminal``, and those listed on https://docs.ansible.com/ansible/devel/plugins/plugins.html  This list can be verified by looking at the last element of the package argument of each ``*_loader`` in https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/loader.py#L1126
  :plugin_utils: For shared code which is only used controller-side, not in modules.

>   :sub_plugins: For other plugins which are managed by modules inside of collections instead of ansible-core.  We use a subfolder so there aren't conflicts when ansible-core adds new plugin types.

Reviewer: ```suggestion
  :sub_plugins: For other plugins which are managed by action plugins inside of collections instead of ansible-core.  We use a subfolder so there aren't conflicts when ansible-core adds new plugin types.
```
modules cannot access this folder.

>   :sub_plugins: For other plugins which are managed by plugins inside of collections instead of ansible-core.  We use a subfolder so there aren't conflicts when ansible-core adds new plugin types.

Reviewer: If we think of modules as a kind of plugin, this wording suggested by @mattclay is better:
```suggestion
  :sub_plugins: For other plugins which are managed by non-module plugins inside of collections instead of ansible-core.  We use a subfolder so there aren't conflicts when ansible-core adds new plugin types.
```

Author: We'll do this as a separate PR.

>   :sub_plugins: For other plugins which are managed by action plugins inside of collections instead of ansible-core.  We use a subfolder so there aren't conflicts when ansible-core adds new plugin types.

Reviewer: ```suggestion
  :sub_plugins: For other plugins which are managed by non-module plugins inside of collections instead of ansible-core.  We use a subfolder so there aren't conflicts when ansible-core adds new plugin types.
```
